isis/tests/FunctionalTestsOverlapstats.cpp

```diff
@@ -0,0 +1,259 @@
+  QVector<QString> fio_args = { "OVERLAPLIST=" + overlapsPath };
+  UserInterface fioOptions(FIO_APP_XML, fio_args);
+  findimageoverlaps(cubes, fioOptions);
+
+  QVector<QString> args = { "FROMLIST=" + cubeListPath,
+                            "OVERLAPLIST=" + overlapsPath };
+  UserInterface options(APP_XML, args);
+  Pvl appLog;
+  overlapstats(options, &appLog);
+
+  PvlGroup group = appLog.findGroup("Results");
+  EXPECT_EQ( (double) group.findKeyword("ThicknessMinimum"), 0.37266300670341);
+  EXPECT_EQ( (double) group.findKeyword("ThicknessMaximum"), 0.37266300670341);
+  EXPECT_EQ( (double) group.findKeyword("ThicknessAverage"), 0.37266300670341);
+  EXPECT_EQ( (double) group.findKeyword("ThicknessStandardDeviation"), -1.79769313486231e+308);
+  EXPECT_EQ( (double) group.findKeyword("ThicknessVariance"), -1.79769313486231e+308);
+  EXPECT_EQ( (double) group.findKeyword("AreaMinimum"), 83798250265.466);
+  EXPECT_EQ( (double) group.findKeyword("AreaMaximum"), 83798250265.466);
+  EXPECT_EQ( (double) group.findKeyword("AreaAverage"), 83798250265.466);
+  EXPECT_EQ( (double) group.findKeyword("AreaStandardDeviation"), -1.79769313486231e+308);
+  EXPECT_EQ( (double) group.findKeyword("AreaVariance"), -1.79769313486231e+308);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackMinimum"), 2.0);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackMaximum"), 2.0);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackAverage"), 2.0);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackStandardDeviation"), -1.79769313486231e+308);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackVariance"), -1.79769313486231e+308);
+  EXPECT_EQ( (int) group.findKeyword("PolygonCount"), 2);
+}
+
+
+TEST_F(ThreeImageNetwork, FunctionalTestOverlapstatsNoOverlap) {
+  // create footprint for cube3 that has no overlap with cubes 1 and 2
+  lonLatPts = new geos::geom::CoordinateArraySequence();
+  lonLatPts->add(geos::geom::Coordinate(50, 50));
+  lonLatPts->add(geos::geom::Coordinate(50, 40));
+  lonLatPts->add(geos::geom::Coordinate(40, 40));
+  lonLatPts->add(geos::geom::Coordinate(40, 50));
+  lonLatPts->add(geos::geom::Coordinate(50, 50));
+
+  polys = new std::vector<geos::geom::Geometry *>;
+  poly = globalFactory->createPolygon(globalFactory->createLinearRing(lonLatPts), nullptr);
+  polys->push_back(poly->clone());
+  multiPoly = globalFactory->createMultiPolygon(polys);
+
+  geos::io::WKTWriter *wkt = new geos::io::WKTWriter();
+
+  std::string polyStr = wkt->write(multiPoly);
+  int polyStrSize = polyStr.size();
+  std::istringstream polyStream(polyStr);
+
+  Blob pvlBlob("Footprint", "Polygon");
+  Pvl pvl;
+  PvlObject polyObject = PvlObject("Polygon");
+  polyObject.addKeyword(PvlKeyword("Name", "Footprint"));
+  polyObject.addKeyword(PvlKeyword("StartByte", "1"));
+  polyObject.addKeyword(PvlKeyword("Bytes", toString(polyStrSize)));
+  pvl.addObject(polyObject);
+
+  pvlBlob.Read(pvl, polyStream);
+  cube3->write(pvlBlob);
```

Comment:
Would it be easy enough to replace this with the cleaner version in the fixture? 

e.g: 
```C++
ImagePolygon poly;
    coords = {{30, 0},
              {30, 10},
              {35, 10},
              {35, 0},
              {30, 0}};
poly.Create(coords);
cube1->write(poly);
cube1->reopen("rw");
```

Reply:
Not sure where you're seeing this, & I'm unable to get such a thing to work since ImagePolygon::Create() takes much different parameters. What I'm doing here seems to be the convention based on whats in the fixture and other tests using ThreeImageNetwork.

Reply:
I'm seeing it on the fixtures for three image adjacency: https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/tests/Fixtures.cpp#L186 

If it's not easily do-able you can not worry about it

Reply:
I stand corrected, that's some of adams work that was merged last week. Looks like I could totally do that and it's way cleaner.